### ISSUE TITLE
setting closurePath

### ISSUE THREAD

**Issue by charleshross** [gmarty/grunt-closure-compiler#38]
if 'closurePath' is a direct filepath to the closure jar file you should make it so it can use that instead of concatenating /build/compiler.jar at the end all the time.

**Comment by langdonx** [comment]
+1

**Comment by mr-moon** [comment]
+1

**Comment by langdonx** [comment]
I saw this, but I don't think it's a very good solution:
- https://github.com/gmarty/grunt-closure-compiler/pull/36

A think a better solution would be...
- if file exists closurePath (file exists, not directory), use it
- else if file exists closurePath + "/compiler.jar", use it
- else if file exists closurePath + "/build/compiler.jar", use it
- else throw an error

**Comment by software-engineering** [comment]
+1

**Comment by jhaoda** [comment]
+1

**Comment by quantuminformation** [comment]
+1

**Comment by gmarty** [comment]
I agree the current situation is not ideal. The best approach would be to ship `grunt-closure-compiler
` with Closure Compiler. So maybe something like a global npm install that would download Closure Compiler and the externs definition files once for all.
I don't really have time to work on that at the moment, but I'm happy to accept PR.

**Comment by ChadKillingsworth** [comment]
:+1:

**Comment by ChadKillingsworth** [comment]
@gmarty Closure compiler is now officially published to npm. Perhaps the compiler should be listed as a node peer dependency? That would allow users to control the version of the compiler separately from the grunt plugin.

https://www.npmjs.com/package/google-closure-compiler

**Comment by ChadKillingsworth** [comment]
Apparently `peerDependencies` are being deprecated, so that option probably isn't the best.

**Comment by mandrews** [comment]
+1

**Comment by ChadKillingsworth** [comment]
The official distribution of closure-compiler has a grunt and gulp plugin included.
